astropy/units/tests/test_quantity.py

```diff
@@ -988,6 +988,80 @@ def test_to_string(self):
             f"Quantity as KMS: {qscalar.to_string(precision=3, unit=u.km / u.s)}" == res
         )
 
+        # Scientific notation with 2 decimal places
```

Comment:
I think this may be a bit more readable as a separate test method that is parametrized. I'd also just test the output of `to_string()`, not its use inside an f-string.

Indeed, I'd use the same parametrization to also check that the output for different formats is OK (latex, unicode), i.e., something like,
```
@pytest.mark.parametrize("q, generic, latex, unicode", [
    (1.5e14 * u.m / u.s, "1.50e+14 m / s", "...", "..."),
    (0.1234 * u.m / u.s, ...),
    ...
])
def test_format_spec(self, q, generic, latex, unicode):
    assert q.to_string(format_spec=format_spec) == generic
    assert q.to_string('latex', format_spec=format_spec) == latex
    ...
```

Reply:
My intention with this comment was to be sure that we test the output for all three relevant formats for all examples. I still think that is a good idea, since the three go through different paths. Hence, I unresolved this...